### DISCUSSION TITLE
NextLink use with  material ui link Hydration failed

### DISCUSSION THREAD

**Discussion by abmarrero** [vercel/next.js#58598] [Help]
### Summary

            
             <NextLink href='./' passHref>
                <Link display='flex' alignItems='center' >

                    <Typography variant='h6' >Clone |</Typography>
                    <Typography sx={{ml: 0.5}}>Shop</Typography>    
                   </Link> 
              </NextLink>

**Comment by icyJoseph** [comment]
I guess, you are having an invalid DOM issue? which then turns into a hydration error? 

The solution is to read the docs from Material UI's Link, and notice that, like many other libraries, you can pass a custom Component. In this case it is a prop called `component`:

- https://mui.com/material-ui/api/link/#props

See this example: https://codesandbox.io/s/cocky-shaw-3ng4dp?file=/src/Demo.tsx, where:

```tsx
/* eslint-disable jsx-a11y/anchor-is-valid */
import * as React from "react";
import Box from "@mui/material/Box";
import Link from "@mui/material/Link";

const preventDefault = (event: React.SyntheticEvent) => event.preventDefault();

const MyLink = React.forwardRef<
  HTMLAnchorElement,
  React.ComponentPropsWithoutRef<"a">
>(({ onClick, href, children }, ref) => {
  return (
    <a href={href} onClick={onClick} ref={ref}>
      Click Me {children}
    </a>
  );
});

export default function Links() {
  return (
    <Box
      sx={{
        typography: "body1",
        "& > :not(style) ~ :not(style)": {
          ml: 2
        }
      }}
      onClick={preventDefault}
    >
      <Link href="#" component={MyLink}>
        Link
      </Link>
    </Box>
  );
}
```

In your case, `MyLink` would be the `next/link`.
